homeassistant/components/media_player/mpd.py

```diff
@@ -142,12 +142,15 @@ def media_duration(self):
     def media_title(self):
         """ Title of current playing media. """
         name = self.currentsong.get('name', None)
-        title = self.currentsong['title']
+        title = self.currentsong.get('title', None)
 
         if name is None:
             return title
         else:
-            return '{}: {}'.format(name, title)
+            if title is None:
```

Comment:
I am not 100% sure that i understand you, but while looking at the code i noticed when 'name' is None, you would get something like 'None: $songtitle'. I will change my pull request so that it all the situations can be handled.